### PR TITLE
Remove int/uint in libcore/lib.rs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #22240.

### CODE REVIEW

**Comment by aturon** [comment]
The actual file defining the module should be removed as well.

**Comment by GuillaumeGomez** [comment]
Since the last commit does the exact same thing as #22503, I think we can close #22503.

File: src/librand/rand_impls.rs [hunk 1]
```diff
@@ -12,18 +12,18 @@
...
     #[inline]
-    fn rand<R: Rng>(rng: &mut R) -> int {
-        if int::BITS == 32 {
-            rng.gen::<i32>() as int
+    fn rand<R: Rng>(rng: &mut R) -> isize {
+        if isize::BITS == 32 {
+            rng.gen::<i32>() as isize
         } else {
-            rng.gen::<i64>() as int
+            rng.gen::<i64>() as isize
```
Comment (GuillaumeGomez) [inline src/librand/rand_impls.rs]: @aturon: I kept the same logic, but I'm not sure it's correct. What's your opinion on this ?

File: src/librand/rand_impls.rs [hunk 2]
```diff
@@ -56,13 +56,13 @@ impl Rand for i64 {
...
     #[inline]
-    fn rand<R: Rng>(rng: &mut R) -> uint {
-        if uint::BITS == 32 {
-            rng.gen::<u32>() as uint
+    fn rand<R: Rng>(rng: &mut R) -> usize {
+        if usize::BITS == 32 {
+            rng.gen::<u32>() as usize
         } else {
-            rng.gen::<u64>() as uint
+            rng.gen::<u64>() as usize
```
Comment (GuillaumeGomez) [inline src/librand/rand_impls.rs]: @aturon: Same here.

**Comment by GuillaumeGomez** [comment]
@aturon ? Your opinion here please ?

**Comment by alexcrichton** [comment]
@bors: r-

Er actually the `pub use isize as int;` annotations in `libstd/lib.rs` also need to be removed I believe.

**Comment by GuillaumeGomez** [comment]
Done !

**Comment by alexcrichton** [comment]
(forgot a push?)

**Comment by GuillaumeGomez** [comment]
My last commit did or maybe I misunderstood you... What line where you talking about ?

**Comment by dotdash** [comment]
@alexcrichton for some reason github displays the new commit above your r+ comment, although it is newer than that

**Comment by Manishearth** [comment]
This one fails tests from `libcoretest` :/

**Comment by GuillaumeGomez** [comment]
I fixed tests. @alexcrichton @Manishearth

**Pull request merged**